site/src/pages/WorkspacePage/Workspace.tsx

```diff
@@ -170,175 +181,217 @@ export const Workspace: FC<React.PropsWithChildren<WorkspaceProps>> = ({
-              severity="error"
-              actions={
-                <Button
-                  onClick={
-                    canRetryDebugMode ? handleBuildRetryDebug : handleBuildRetry
-                  }
-                  variant="text"
-                  size="small"
-                >
-                  Retry{canRetryDebugMode && " in debug mode"}
-                </Button>
-              }
-            >
-              <AlertTitle>Workspace build failed</AlertTitle>
-              <AlertDetail>{workspace.latest_build.job.error}</AlertDetail>
-            </Alert>
-          )}
-
-          {template?.deprecated && (
-            <Alert severity="warning">
-              <AlertTitle>Workspace using deprecated template</AlertTitle>
-              <AlertDetail>{template?.deprecation_message}</AlertDetail>
-            </Alert>
-          )}
-
-          {transitionStats !== undefined && (
-            <WorkspaceBuildProgress
-              workspace={workspace}
-              transitionStats={transitionStats}
-            />
-          )}
-
-          {buildLogs}
-
-          {typeof resources !== "undefined" && resources.length > 0 && (
-            <Resources
-              resources={resources}
-              agentRow={(agent) => (
-                <AgentRow
-                  key={agent.id}
-                  agent={agent}
-                  workspace={workspace}
-                  sshPrefix={sshPrefix}
-                  showApps={canUpdateWorkspace}
-                  showBuiltinApps={canUpdateWorkspace}
-                  hideSSHButton={hideSSHButton}
-                  hideVSCodeDesktopButton={hideVSCodeDesktopButton}
-                  serverVersion={buildInfo?.version || ""}
-                  serverAPIVersion={buildInfo?.agent_api_version || ""}
-                  onUpdateAgent={handleUpdate} // On updating the workspace the agent version is also updated
-                />
-              )}
-            />
-          )}
-
-          {workspaceErrors.getBuildsError ? (
-            <ErrorAlert error={workspaceErrors.getBuildsError} />
-          ) : (
-            <BuildsTable
-              builds={builds}
-              onLoadMoreBuilds={onLoadMoreBuilds}
-              isLoadingMoreBuilds={isLoadingMoreBuilds}
-              hasMoreBuilds={hasMoreBuilds}
-            />
-          )}
-        </Stack>
-      </Margins>
-    </>
+            {template?.deprecated && (
+              <Alert severity="warning">
+                <AlertTitle>Workspace using deprecated template</AlertTitle>
+                <AlertDetail>{template?.deprecation_message}</AlertDetail>
+              </Alert>
+            )}
+
+            {transitionStats !== undefined && (
+              <WorkspaceBuildProgress
+                workspace={workspace}
+                transitionStats={transitionStats}
+              />
+            )}
+
+            {buildLogs}
+
+            {typeof resources !== "undefined" && resources.length > 0 && (
```

Comment:
Could the condition be simplified to `resources?.length > 0`?

Reply:
No, because typescript will throw an error:

<img width="694" alt="Screenshot 2024-01-05 at 11 59 55" src="https://github.com/coder/coder/assets/3165839/6eb52286-ba1e-45f0-8a3a-eff14137e2df">

Reply:
Oh, that's my mistake – this is one of those things that's technically safe in JavaScript (`undefined > 0` evaluates to `false`), but TS wouldn't be happy